control-plane/csi-driver/controller/src/controller.rs

```diff
@@ -398,9 +380,16 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
                     }
                 },
             _ => {
+                // if the csi-node happens to be a data-plane node, use that for nexus creation, otherwise
+                // let the control-plane select the target node.
+                let target_node = match IoEngineApiClient::get_client().get_node(&node_id).await {
+                    Ok(_) => Some(node_id.as_str()),
```

Comment:
this is not quite accurate, depends on the type of error? Eg: we could simply fail to issue the api call but that doesn't mean the node is not a data-plane node.

I think we should also discuss what to do here with a wider audience:
Options:
1. Always defer target placement to the control-plane.
2. Run on app node is possible.

Reply:
That's right, but anyhow if we are not able to determine it is a `dataplane` node or not we will always end up with Option 1. So probably option 1 is the right way.

Reply:
We can determine if it's a `dataplane` by inspecting checking the exact error we receive? If it's 404 then it must not be a `dataplane` node?

Reply:
Done.